### ISSUE TITLE
Unrealistic shifts in the Central Arctic

### ISSUE THREAD

**Issue by NicWayand** [hdirector/IceCastV2#2]
@hdirector @cmbitz I am seeing some strange "jagged" artifacts in the bias corrections. See below comparison plot (10 day lead time), which shows a very jagged contour in Central Arctic. This prediction was trained on a 30 day period, with a 10 day gap before the initialization time (2017-09-03). Since the model here (CFSv2) does quite well, I expected the corrections to be much smaller. They appear to be very large, even removing all ice until the center origin for the [25 day lead time](https://github.com/NicWayand/stcs/blob/master/Compare_CS_output.ipynb). So maybe this is indicating a bug in my code [changes](https://github.com/hdirector/IceCastV2/compare/master...NicWayand:short)? Do you have a built in 

![image](https://user-images.githubusercontent.com/1117224/41869250-49ef1cc6-786d-11e8-83a3-c392fae8d7c0.png)

Full analysis code and plots here:
https://github.com/NicWayand/stcs/blob/master/Compare_CS_output.ipynb

**Comment by hdirector** [comment]
Could you point me to the R code that goes along?  I'm only seeing the Python code at your link. I have a guess what's wrong, but can't tell for sure with only the Python calls.

**Comment by NicWayand** [comment]
This link (https://github.com/hdirector/IceCastV2/compare/master...NicWayand:short) shows the changes I have made (left is original code, right is my changes)

The main change is here (https://github.com/hdirector/IceCastV2/compare/master...NicWayand:short#diff-02cb6d42dc6d7b8fc9af1ce3c603c0d2L99) where I have replaced he `- 1` (year before current prediction year) with ` - endYearOffset` (a user defined offset before current prediction year)

**Comment by NicWayand** [comment]
It appears to be happening near large CFSv2 model discontinuities (low model resolution). Maybe the deltas are calculated and applied across these, resulting in the large changes? Is it easy to increase the number of Central Arctic lines `centLines`? (I didn't see any functions for calculating the regionInfo.R.Data file).

**Comment by hdirector** [comment]
Did you set predYears to the day index and month to 1 when calling quickRun? My first guess based on the output is that somewhere the value that's being given to bcYear in the ContourShift() function is still being passed the index of the year rather than the day. So, you're predicting something like the gap at day 2017 not day 40 (which would be much bigger than we want)

It's possible to increase the number of centLines, but doing so may create other problems. The number of lines is currently set to match the number of points on the corners of the grid box. So, if you don't have enough lines to match the coastlines, you won't actually be able to get ice up to it accurately.

**Comment by NicWayand** [comment]
The obsNCDF and predNCDF files I created look like this:

![image](https://user-images.githubusercontent.com/1117224/41874700-7948effa-787d-11e8-859e-c4ce774a05ea.png)

where years is:

![image](https://user-images.githubusercontent.com/1117224/41874740-908dbe3e-787d-11e8-9b9f-b528c6d712cc.png)

So their index is equal to their "year" value since R is base 1. 

Below is the call to quckRun I am doing:
```r
quickRun(obsNCDF = "Obs_test.nc", 
             predNCDF = "Mod_test.nc", 
             predYears = 244, 
             startYear = 242, 
             endYearOffset = 1,
             month = 1, 
             outputFile = "temp.nc", 
             level = 15,
             datTypeObs = "bootstrap",
             plotting = TRUE)
```

I checked the value going into the rlm [here](https://github.com/hdirector/IceCastV2/compare/master...NicWayand:short#diff-02cb6d42dc6d7b8fc9af1ce3c603c0d2L109) and I am fairly sure they are correct, for example on:
`lmXObs <- suppressWarnings(rlm(xTempObs ~ yearInd))`
yearInd = 242 243
and in:
`xObsPred <- predict(lmXObs, newdata = data.frame(yearInd = bcYear))`
bcYear = 244
Which as I understand the code, is fitting the linear model to "years" 242 and 243, and retrieving the predicted value at 244.

Here are the two generated plots of the central arctic:
![image](https://user-images.githubusercontent.com/1117224/41877070-d1fd31ae-7884-11e8-8c52-e5ad364e10df.png)
![image](https://user-images.githubusercontent.com/1117224/41877108-f83a8c18-7884-11e8-89d2-3ba350bd306c.png)

**Comment by hdirector** [comment]
Apologies if you've accounted for this elsewhere and I'm not just seeing it. But it's looks like from the ncdump your `conc` variable is dimension lat x lon x month x year. However,  `quickRun` is expecting dimension year x month x lon x lat. This arrangement matters in this line, for example:
```
 discrep <- createMapping(startYear = startYear, endYear = max(predYears) - endYearOffset,
                           obsStartYear = obsStartYear, predStartYear = predStartYear,
                           observed = obsMat[,month,,], predicted = predMat[,month,,],
                           regions = regionInfo, month = month, level = level,
                           datTypeObs = datTypeObs, datTypePred = "simple", plotting = plotting)
```
If this doesn't fix things, you're welcome to send me the obsNCDF and predNCDF files and I can look into what's going on (although I probably won't have time until Wednesday.)

**Comment by NicWayand** [comment]
I think I am accounting for this. I formatted the netcdf file to match your example obs file. When R reads the file in conc dims look like `365 43 304 448 (years, months, lon, lat)` (I don't know why R switches them around), so it looks like it is properly formatted for createMapping().

**Comment by hdirector** [comment]
Makes sense. Now that you mention, I've run into this before where R rearranges ncdf files. 

Sounds good.
